src/shared/parameter_v2/config/example.yaml

```diff
@@ -0,0 +1,75 @@
+# Examples of all supported Parameter types and their schemas
+--- # Metadata for this config
+metadata:
+  # If shared is true, all parameters in this file will be visible in configs at the same
+  # level in the directoy structure, as well as all children.
+  shared: false
+  # The name of this config. This will be the name used in the generated code
+  name: ExampleConfig
+--- # Parameters
+BoolParam:
+  type: bool
+  value: true
+  description: >
+    Can be true or false
+DoubleParam:
+  type: double # could also be float
+  min: 0
+  max: 5
+  value: 0.2
+  description: >
+    Can be any value in the range [min, max]
+IntParam:
+  type: int # could also be unsigned
```

Comment:
Same as above

Reply:
I think this makes more sense, there is a semantic difference to signed vs unsigned numbers, `double` vs `float` is really just about precision.

Reply:
Yea same response as above, just me being lazy. In theory we could use any integral type here: `int_16`, `uint_8` but I also dont' think we generally have a use-case for those, so may be better to allow only things explicitly listed in the `example` file.

Thoughts on generically accepting int types vs being very explicit about what's specifically allowed?

Reply:
IMO we should be explicit because that allows to control the types used to not use "non-standard" types and give the user more informative error messages at generation time, rather then fail at compile time.

Reply:
yup I agree. Will update the schema

Reply:
Updated the example to explicitly list all valid parameter types.